site/content/blog/2023-11-03-handling-modern-web-apps-better-part1/index.md

```diff
@@ -0,0 +1,118 @@
+
+ZAP can explore modern apps pretty effectively using the [AJAX Spider](/docs/desktop/addons/ajax-spider/) and can find [DOM XSS](/docs/desktop/addons/dom-xss-active-scan-rule/) vulnerabilities by launching browsers. But we know that we could still do better, and based on our recent [What Should We Focus On](/blog/2023-08-29-what-should-we-focus-on/) questionnaire we know it is your top priority too.
+
+So we have a brand new [Client Side Integration](/docs/desktop/addons/client-side-integration/) add-on (also called the Client add-on for short) which helps ZAP understand what is going on inside the browser.
+
+### ZAP Chat Video
+
+{{<youtube uuid="Rq_d7OLmMfw">}}
+
+Staring [OWASP Juice Shop](https://owasp.org/www-project-juice-shop/)
+
+### Client Visibility
+
+The [Client](/docs/desktop/addons/client-side-integration/) add-on is not currently included in ZAP by default so you will need to add it via the [ZAP Marketplace](/addons/).
+ 	
+The add-on adds 3 new tabs to ZAP which give you a lot more information about what is happening in the browser.
+
+All of this information is available via Browser dev tools, but only for the pages being displayed.
+ZAP shows you all of the essential details for the whole app, and this information is also available when automating ZAP.
+
+#### Client Map
+
+The [Client Map](/docs/desktop/addons/client-side-integration/#client-map) is a representation of the apps that the browser accesses, so superficially it looks similar to the [Sites Tree](/docs/desktop/start/features/sitestree/).
+
+However if you look closely you will see that it actually includes the [URL fragments](https://en.wikipedia.org/wiki/URI_fragment) which are not sent from the browser to the target app, so ZAP doesnt not normally see these.
+These fragments give you a much more detailed understanding of the client side structure of the app.
+
+![Client Map and Details Screenshot](images/client-map-details.png)
+
+The Client Map includes URLs that have not yet been visited - these have a small red ‘minus’ sign to indicate URLS that have been found in the DOM but which have not been directly accessed by ZAP.
+
+The Client Map also includes storage nodes so you can see the contents of cookies, local storage, and session storage in the browser.
+
+![Client Storage Screenshot](images/client-storage.png)
+
+It supports a set of context menu options - these can be accessed via a "right click" or the relevant key combination for your OS.
+
+#### Client Details
+
+The [Client Details](/docs/desktop/addons/client-side-integration/#client-details) tab (also shown above) shows details about the node selected in the Client Map.
+
+It allows you to understand much more about the functionality that this URL represents in the target app.
+
+The types of data displayed include:
+
+* Button: Buttons detected in the DOM
+* Cookies: Cookies set in the browser
+* FORM: Forms detected in the DOM
```

Comment:
I don't have a quick way to check right this minute but is FORM really full caps vs the others?

Purposefully?

Reply:
Yes, thats how it appears in the GUI. Can't remember why off hand, but to change that would be a code change..

Reply:
Alright leave it for now :wink: